### DISCUSSION TITLE
qute-pass: no gpg password prompt

### DISCUSSION THREAD

**Discussion by fritzrehde** [qutebrowser/qutebrowser#7082] [Q&A] [answered]
I want to use the qute-pass script to auto-fill passwords using pass. However, if I haven't recently unlocked the gpg key (I am guessing the password gets cached), qute-pass will only autofill the username (which obviously makes sense from a security standpoint since I haven't entered my gpg password). However, I thought I would get some prompt to input my gpg password if it isn't cached, but this is not the case. Is there a way to achieve the desired behaviour? Could this have something to do with pinentry (I normally use the terminal based ncurses version, maybe it doesn't work with qutebrowser). The command/mapping I use is `:spawn --userscript qute-pass`.
Thank you!

**Comment by fritzrehde** [comment, marked as answer]
I have found the answer to my question: Pinentry was the problem, asking for the password does not work if pinentry-ncurses is the pinentry version used, I am now using pinentry-gnome3 which works fine.

**Reply by piperfw** [reply]
Note pinentry-gnome3 has the gnome3-backend `gcr` as an additional dependency. If that is not on your system you can try the gtk2 or qt versions (requires `gtk2`, `qt5-base` respectively). See `ls /usr/bin/pinentry-*` and how to set the default version on the [arch wiki](https://wiki.archlinux.org/title/GnuPG#pinentry).

**Reply by ghost** [reply]
Hi, could you please show what configuration file did you change to achieve it? I'm using Sway compositor over Wayland and I'm not pretty sure which would be the best approach in order to obtain a prompt for my password when using the userscript qute-pass.

**Reply by piperfw** [reply]
Hi, I believe it is `~/.gnupg/gpg-agent.conf` with the line e.g. `pinentry-program /usr/bin/pinentry-qt` (see the wiki page I linked for more information).

**Comment by piperfw** [comment]
I wanted to be able to still use `pinentry-ncurses` when working from the Shell (and I only had the requirements for `pinentry-qt5`) so taking inspiration from [this solution](https://unix.stackexchange.com/a/613935/417917), I
1. Created a wrapper `~/userscripts/pinentry-wrapper` 
```
#!/usr/bin/env bash
# Defaults to Qt, with a choice of curses for selected programs
case "$PINENTRY_USER_DATA" in
    curses)
        exec /usr/bin/pinentry-curses "$@"
        ;;
    *)
        exec /usr/bin/pinentry-qt "$@"
        ;;
esac
```
and made this executable.
2. Added to `~/.gnupg/gpg-agent.conf`
```
pinentry-program ~/userscripts/pinentry-wrapper
```
(force restart the agent `pkill -HUP gpg-agent` after this).
3. Added `export PINENTRY_USER_DATA=curses` to my shell startup files e.g. `~.zshrc`
